Add contribution guidelines (#31)
sa-tre/satre-specification: docs/source/contributing.md

The community can suggest governance changes at any point.
This includes the SATRE team, and any decision must be openly documented in the repo.
:::

### Specification Format

The specification source is kept in the [specification repository](https://github.com/sa-tre/satre-specification).
It is written as a [Sphinx](https://www.sphinx-doc.org/) document in [Markdown](https://www.markdownguide.org/) format.

The most up-to-date 'source of truth' will be the specification on the `main` branch of the specification repository.
The community can decide when to 'tag' a new version of the specification.
They may also decide to where to publish the specification.

The specification repository is self-contained and relates only to the specification specifically, or its governance.
Any contributions to the wider SATRE project should be made through a different medium via the [SATRE GitHub organisation](https://github.com/sa-tre), or by contacting the SATRE team at [satre-contact@dundee.ac.uk](mailto:satre-contact@dundee.ac.uk).

### Contribution Process

Issues should be used to discuss ideas, potential changes and to ask questions.

> Issue templates have been designed for common issue archetypes to help collect the most important information and present it in a clear, consistent way.

Reviewer: ```suggestion
Issue templates have been designed for common issue types to help collect the most important information and present it in a clear, consistent way.
```
I think simple English is best unless "archetypes" conveys something special?

Author: Absolutely, as I was writing this I was thinking "Surely I can think of a better word than archetypes" :smile:

It is possible, however, to open a blank issue if none of the templates are suitable.

When ready, changes will be proposed in pull requests.
Similarly to issues, there is a pull request template.
This template prompts contributors to include important details which helps explain the contribution and makes triage and review easier.

Pull requests will be used to review changes.
During the review process, the pull request will be used for discussion, to suggest amendments and ultimately accept or reject the change.

We use this process to ensure that as much as possible of the discussion and decision making process can be public.
This is to provide as open and accessible as possible an environment for all contributors to engage in the conversation.

### Consensus Mechanism

Approval from the SATRE team is based on [lazy consensus](https://medlabboulder.gitlab.io/democraticmediums/mediums/lazy_consensus/).

> Once a pull request has been open for 7 days, unless there are outstanding objections within the comments, the SATRE team will approve the change.

Reviewer: ```suggestion
Once a pull request has been open for at least 7 days, unless there are outstanding objections within the comments, the SATRE team may approve the change at any point.
```
Otherwise this implies it will be merged after 7 days, when we want more freedom to wait.

Reviewer: Yes, I'm happy with any wording that doesn't imply we will merge immediately after 7 days.

Author: Updated in [4c4f5a4](https://github.com/sa-tre/satre-specification/pull/31/commits/4c4f5a466d6e9af9e595f9f02bbd128a507b1e61)

If any of the SATRE team objects to a particular issue, this will be raised by individuals in the comments of the pull request.
The SATRE team will only approve pull requests that have no outstanding objections.

## Writing in Markdown

The [Myst Parser](inv:myst-parser:std:doc#syntax/typography) documentation has a guide on the Markdown format used in the specification source files.
GitHub also has a helpful page on [getting started with writing and formatting on GitHub](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github), which will be useful when writing Markdown for GitHub (for example in issue or pull request comments).

You can think of Markdown as a few little symbols around your text that instruct how to render the text.

> For example, you could write words in **bold** (`**bold**`), in _italics_ (`_italics_`), or as a [link](https://youtu.be/dQw4w9WgXcQ) (`[link](https://youtu.be/dQw4w9WgXcQ)`) to another web page.

Reviewer: 🤔

Author: I assume that is not approval 😄. Any suggestions?

This is mostly lifted from the draft in #5. I made some formatting changes and added the note above about the Myst Markdown syntax (as most of the content should use that rather than GitHub Markdown).

I'm not particularly attached to this. Not sure if the link sets the right tone 🙂.I think we need something to explain what Markdown is (and why we use it?). I think we should be careful not to try and write an introduction to Markdown or a Markdown guide.

To report Code-of-Conduct violations, please use the contact [specified in the Code of Conduct](https://github.com/sa-tre/satre-specification/blob/main/CODE_OF_CONDUCT.md).

## Recognising Contributions

We welcome and recognise all kinds of contributions, from discussing ideas, suggesting features, improving governance, maintaining the project, and more.

This project follows the [All Contributors specification](https://allcontributors.org/docs/en/overview).
The All Contributors bot usage is described [here](https://allcontributors.org/docs/en/bot/usage).

To add yourself or someone else as a contributor, comment on the relevant Issue or Pull Request with the following:

```text
@all-contributors please add <username> for <contributions>
```

You can see the [Emoji Key (Contribution Types Reference)](https://allcontributors.org/docs/en/emoji-key) for a list of valid `<contribution>` types.
The bot will then create a Pull Request to add the contributor and reply with the pull request details.

:::{hint}
**Please only add one contributor with the bot at a time!**

It is best to add each contributor in turn and merge the pull request before adding another one.

> Otherwise, you can end up with merge conflicts.

Reviewer: It's a shame the bot can't automatically rebase and force push.

After a pull request has been open for at least 7 days, unless there are outstanding objections, the change will be presumed as accepted.
SATRE team members are then free to merge the pull request at any point.

If any of the SATRE team objects to a particular issue, this will be raised by individuals using [pull request reviews](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews).
The SATRE team will only merge pull requests that have no outstanding objections.

## Writing in Markdown

The [Myst Parser](inv:myst-parser:std:doc#syntax/typography) documentation has a guide on the Markdown format used in the specification source files.
GitHub also has a helpful page on [getting started with writing and formatting on GitHub](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github), which will be useful when writing Markdown for GitHub (for example in issue or pull request comments).

You can think of Markdown as a few little symbols around your text that instruct how to render the text.

> For example, you could write words in **bold** (`**bold**`), in _italics_ (`_italics_`), or as a [link](https://youtu.be/dQw4w9WgXcQ) (`[link](https://youtu.be/dQw4w9WgXcQ)`) to another web page.

Reviewer: ```suggestion
For example, you could write words in **bold** (`**bold**`), in _italics_ (`_italics_`), or as a [link](https://medium.com/satre) (`[link](https://medium.com/satre)`) to another web page.
```
